test/ZNSRegistry.test.ts

```diff
@@ -459,20 +449,151 @@ describe("ZNSRegistry Tests", () => {
+
+      const [
+        ownerStatusAfter,
+        operatorStatusAfter,
+        existanceAfter,
+        accessControllerAfter,
+      ] = await registryPromises(zns.registry, domainHash, [deployer, operator]);
+
+      expect(ownerStatusBefore).to.eq(ownerStatusAfter);
+      expect(operatorStatusBefore).to.eq(operatorStatusAfter);
+      expect(existanceBefore).to.eq(existanceAfter);
+      expect(accessControllerBefore).to.eq(accessControllerAfter);
+    });
+
+    it("Validates new logic after upgrade", async () => {
+      const registryFactory = new ZNSRegistryMock__factory(deployer);
+
+      // Add an operator
+      await zns.registry.connect(deployer).setOwnerOperator(operator.address, true);
+
+      // Create a domain record
+      const domainHash = hashDomainName("world");
+      await zns.registry.connect(mockRegistrar).createDomainRecord(
+        domainHash,
+        deployer.address,
+        mockResolver.address
+      );
+
+      // Get original values to ensure they're not modified or overwritten
+      const [
+        ownerStatusBefore,
+        operatorStatusBefore,
+        existanceBefore,
+        accessControllerBefore,
+      ] = await registryPromises(zns.registry, domainHash, [deployer, operator]);
+
+      // TODO upgrade this way for all the contracts
+      await hre.upgrades.upgradeProxy(
+        zns.registry.address,
+        registryFactory,
+        {
+          kind: "uups",
+        }
+      );
 
-      await expect(tx).to.emit(registry, "DomainRecordDeleted").withArgs(wilderDomainHash);
+      // Accessing directly through `zns.registry` won't provide the updated type
+      // Instead, we attach the new factory to that address after upgrading
+      const upgradedRegistry = registryFactory.attach(zns.registry.address);
+
+      const newNumber = ethers.BigNumber.from("123");
+      await upgradedRegistryActions(upgradedRegistry, newNumber, randomUser);
+
+      const [
+        newMappingCall,
+        newMappingSpecificCall,
+        newNumberCall,
+        newAddressCall,
+      ] = await upgradedRegistryPromises(upgradedRegistry, newNumber);
+
+      expect(newMappingCall).to.eq(randomUser.address);
+      expect(newMappingSpecificCall).to.eq(randomUser.address);
+      expect(newNumberCall).to.eq(newNumber);
+      expect(newAddressCall).to.eq(randomUser.address);
+
+      // Verify original values are still the same
+      const [
+        ownerStatusAfter,
+        operatorStatusAfter,
+        existanceAfter,
+        accessControllerAfter,
+      ] = await registryPromises(zns.registry, domainHash, [deployer, operator]);
+
+      expect(ownerStatusBefore).to.eq(ownerStatusAfter);
+      expect(operatorStatusBefore).to.eq(operatorStatusAfter);
+      expect(existanceBefore).to.eq(existanceAfter);
+      expect(accessControllerBefore).to.eq(accessControllerAfter);
+    });
+
+    it("Fails when an unauthorized account tries to call to upgrade", async () => {
```

Comment:
this tests's title contradicts the test itself

Reply:
I had mistakenly put the negative test case in the positive test and vice versa, I've already reversed this in the latest however